### DISCUSSION TITLE
How can I make a static site with pages?

### DISCUSSION THREAD

**Discussion by chuckadams** [hannoeru/vite-plugin-pages#90] [Q&A] [answered]
When I run `vite build`, it builds my app with working pages ... as long as I only start from the index.html in the root.  I'd assumed when I set `nuxtStyle: true`, that it would behave like nuxt, namely generate the landing pages in the built site so that people could navigate directly to pages from bookmarks in their browser, e.g. https://example.com/app/some/nested/page/here, but all I get is the flat directory in dist

Is this possible with the pages plugin, and if not, any workarounds to suggest?   I really want to keep using Vite with Vue3 and not have to go back to Nuxt.

**Comment by hannoeru** [comment, marked as answer]
Vue default runs SPA mode so you only get one HTML file in the dist folder.
If you want to use SSG to build a static website, try [vite-ssg](https://github.com/antfu/vite-ssg), it can work with this plugin.

Also, suggest you start with [vitesse](https://github.com/antfu/vitesse) it's using both vite-plugin-pages & vite-ssg.

**Reply by chuckadams** [reply]
It's not quite SSG I'm looking for per se, I'm just looking to be able to serve the app from any of the routes when accessed directly.  e.g. if someone has a link to http://example.com/app/admin/reports, then that link will load the page and navigate to the correct route.  I don't need any server-side state or rendering, just the app working from a bookmark.  Nuxt actually code-splits each page, but I don't really care if it just serves the same monolith for each (it's an internal app, not a public site, so there's bandwidth to burn)

I suppose if I tell SSG to not render anything at all and make everything client-only, that would be effectively equivalent.   I'm not entirely sure how to do that, but I'll tinker with it a bit and see.  And BTW, thanks for the link to Vitesse, I'll definitely give that a look :)

**Reply by hannoeru** [reply]
I think what you want is to redirect all routes bask to index.html?
Just search something like `redirect for SPA`, then you will get the answer.

**Reply by chuckadams** [reply]
Actually it looks like Vitesse does exactly what I need and much more.  Thanks a ton for pointing me at at it!  👍 👍
